### ISSUE TITLE
Tasks-assisted project creation and task locking error

### ISSUE THREAD

**Issue by zlavergne** [hotosm/tasking-manager#1549]
I'm getting a similar `null value in column "id" violated not-null constraint` error when I try to create a project and when I try to lock a task on the `tasks-assisted` instance

Project Creation:
```
Project PUT - unhandled error: (psycopg2.IntegrityError) null value in column "id" violates not-null constraint
DETAIL:  Failing row contains (null, 2, 2019-05-07 14:40:45.756587, 2, en, 5423893, 1, f, f, f, null, null, null, null, null, 2019-05-07 14:40:45.738465, null, null, null, 69, 0, 0, 0, null, 0101000020E6100000DBE0A12E9E235BC0C3FB1337D5874340, 0106000020E61000000100000001030000000100000009000000C5D5D3F2662F..., 0, {0,1,2,3}, {0,1,2,3}).
[SQL: 'INSERT INTO projects (status, created, priority, default_locale, author_id, mapper_level, enforce_mapper_level, enforce_validator_role, private, entities_to_map, changeset_comment, due_date, imagery, josm_preset, last_updated, license_id, geometry, centroid, task_creation_mode, mapping_types, organisation_tag, campaign_tag, mapping_editors, validation_editors, total_tasks, tasks_mapped, tasks_validated, tasks_bad_imagery) VALUES (%(status)s, %(created)s, %(priority)s, %(default_locale)s, %(author_id)s, %(mapper_level)s, %(enforce_mapper_level)s, %(enforce_validator_role)s, %(private)s, %(entities_to_map)s, %(changeset_comment)s, %(due_date)s, %(imagery)s, %(josm_preset)s, %(last_updated)s, %(license_id)s, ST_SetSRID(ST_GeomFromGeoJSON(%(ST_GeomFromGeoJSON_1)s), %(ST_SetSRID_1)s), ST_Centroid(ST_SetSRID(ST_GeomFromGeoJSON(%(ST_GeomFromGeoJSON_1)s), %(ST_SetSRID_1)s)), %(task_creation_mode)s, %(mapping_types)s, %(organisation_tag)s, %(campaign_tag)s, %(mapping_editors)s, %(validation_editors)s, %(total_tasks)s, %(tasks_mapped)s, %(tasks_validated)s, %(tasks_bad_imagery)s) RETURNING projects.id'] [parameters: {'status': 2, 'created': datetime.datetime(2019, 5, 7, 14, 40, 45, 756587), 'priority': 2, 'default_locale': 'en', 'author_id': 5423893, 'mapper_level': 1, 'enforce_mapper_level': False, 'enforce_validator_role': False, 'private': False, 'entities_to_map': None, 'changeset_comment': None, 'due_date': None, 'imagery': None, 'josm_preset': None, 'last_updated': datetime.datetime(2019, 5, 7, 14, 40, 45, 738465), 'license_id': None, 'ST_GeomFromGeoJSON_1': '{"type": "MultiPolygon", "coordinates": [[[[-108.7406584805204, 39.1657050592006], [-108.4738645142271, 39.15605723798356], [-108.3397513812735, 39.1 ... (112 characters truncated) ... 0053, 38.94646420941521], [-108.761248624727, 39.01439077669514], [-108.7997597817628, 39.05176863972179], [-108.7406584805204, 39.1657050592006]]]]}', 'ST_SetSRID_1': 4326, 'task_creation_mode': 0, 'mapping_types': None, 'organisation_tag': None, 'campaign_tag': None, 'mapping_editors': [0, 1, 2, 3], 'validation_editors': [0, 1, 2, 3], 'total_tasks': 69, 'tasks_mapped': 0, 'tasks_validated': 0, 'tasks_bad_imagery': 0}]
```

Task lock:
```
Task Lock API - unhandled error: (psycopg2.IntegrityError) null value in column "id" violates not-null constraint 
DETAIL: Failing row contains (null, 2, 10, LOCKED_FOR_MAPPING, null, 2019-05-07 14:45:42.143796, 5423893). 
[SQL: 'INSERT INTO task_history (project_id, task_id, action, action_text, action_date, user_id) VALUES (%(project_id)s, %(task_id)s, %(action)s, %(action_text)s, %(action_date)s, %(user_id)s) RETURNING task_history.id'] 
[parameters: {'project_id': 2, 'task_id': 10, 'action': 'LOCKED_FOR_MAPPING', 'action_text': None, 'action_date': datetime.datetime(2019, 5, 7, 14, 45, 42, 143796), 'user_id': 5423893}]
```

**Comment by arunasank** [comment]
I debugged this with @willemarcel today. Thank you, Wille! 

### TL;DR

The issue was with the postgres versions. The `test.sql` being used as a data dump file for the `assisted` stack was created using postgres Engine 11.2, whereas the engine version on the RDS was 9.5.0

### How did we debug

* We looked at the project schemas on the `test.sql`, and saw that they contained:
```
CREATE SEQUENCE public.projects_id_seq
    AS integer
    START WITH 1
    INCREMENT BY 1
    NO MINVALUE
    NO MAXVALUE
    CACHE 1;
```
* When we extracted the schema from the `assisted` stack's RDS, we did not see this in the schema.
* I tried to run the command on the remote RDS, and got this:
```
ERROR:  syntax error at or near "AS"
LINE 1: CREATE SEQUENCE public.projects_id_seq AS integer START WITH...
```
* On looking up the error, we found https://stackoverflow.com/questions/49208448/error-in-creating-sequences-when-restoring-the-postgresql-database mentioning that the `CREATE SEQUENCE ... AS` command is for newer postgres versions.
* We confirmed this by making sure that the dump file used to create the database on the RDS stack was `postgres` version 11.2, while the version of postgres on the RDS was 9.5.15.
* On updating the `postgres` version to 11.2, the issue was fixed. 

Going to create an issue to standardise the version across our dump files to 9.5.15, since we use this on production as well, now, and going to update this again.

**Comment by arunasank** [comment]
@zlavergne feel free to close this after you confirm that the project creation is working. I managed to create a test project successfully.

**Comment by arunasank** [comment]
Closing here, since I tested again today that this works. Feel free to reopen if you notice any issues! 🍎
